src/main/java/gov/usds/case_issues/services/SneakyCaseService.java

```diff
@@ -0,0 +1,239 @@
+		Specification<SneakyViewEntity> snoozed = (root, query, cb) -> 
+			cb.greaterThanOrEqualTo(root.get("snoozeEnd"), ZonedDateTime.now());
+		mainSpec = mainSpec.and(snoozed);
+		if (snoozeReason.isPresent()) {
+			Specification<SneakyViewEntity> snoozeReasonSpec = (root, query, cb) ->
+				cb.equal(root.get("snoozeReason"), snoozeReason.get());
+			mainSpec = mainSpec.and(snoozeReasonSpec);
+		}
+		return wrapFetched(mainSpec, pageInfo);
+	}
+
+	private List<CaseSummary> wrapFetched(Specification<SneakyViewEntity> mainSpec, Pageable pageInfo) {
+		try {
+			LOG.debug("Starting fetch using {}/{}", mainSpec, pageInfo);
+			Slice<SneakyViewEntity> page = _repo.findAll(mainSpec, pageInfo);
+			LOG.debug("Finished fetch using {}/{}", mainSpec, pageInfo);
+			return page.getContent().stream().map(DelegatingSummary::new).collect(Collectors.toList());
+		} catch (InvalidDataAccessApiUsageException e) {
+			if (e.getCause() instanceof IllegalArgumentException) {
+				throw (IllegalArgumentException) e.getCause();
+			} else {
+				throw e;
+			}
+		}
+	}
+
+	@Override
+	public List<CaseSummary> getPreviouslySnoozedCases(String caseManagementSystemTag, String caseTypeTag,
+			String receiptNumber, DateRange caseCreationRange, int size) {
+		Pageable pageInfo = activeCasePage(size);
+		Specification<SneakyViewEntity> pathSpec = commonSpec(caseManagementSystemTag, caseTypeTag,
+				pageInfo.getSort(), Optional.ofNullable(receiptNumber), Optional.ofNullable(caseCreationRange));
+		Specification<SneakyViewEntity> wasSnoozed = (root, query, cb) -> 
+			cb.lessThan(root.get("snoozeEnd"), ZonedDateTime.now());
+		return wrapFetched(Specification.where(pathSpec.and(wasSnoozed)), pageInfo);
+	}
+
+	private static Pageable activeCasePage(int size) {
+		return PageRequest.of(0, size, DEFAULT_SORT);
+	}
+
+	private static Pageable snoozedCasePage(int size) {
+		return PageRequest.of(0, size, SNOOZE_SORT);
+	}
+
+	private Specification<SneakyViewEntity> commonSpec(
+			String caseManagementSystemTag,
+			String caseTypeTag,
+			Sort sortOrder,
+			Optional<String> pageReference, Optional<DateRange> caseCreationRange) {
+		CasePageInfo path = _translator.translatePath(caseManagementSystemTag, caseTypeTag, pageReference.orElse(null)); 
+		Specification<SneakyViewEntity> fullSpec = pathSpec(path);
+		if (!path.isFirstPage()) {
+			fullSpec = fullSpec.and(pageSpec(path.getCase(), sortOrder));
+		}
+		if (caseCreationRange.isPresent()) {
+			DateRange dateRange = caseCreationRange.get();
+			Specification<SneakyViewEntity> dateSpec = (root, query, cb) -> cb.and(
+				cb.greaterThanOrEqualTo(root.get("caseCreation"), dateRange.getStartDate()),
+				cb.lessThanOrEqualTo(root.get("caseCreation"), dateRange.getEndDate())
+			);
+			fullSpec = fullSpec.and(dateSpec);
+		}
+		return fullSpec;
+	}
+
+	private Specification<SneakyViewEntity> pageSpec(TroubleCase troubleCase, Sort sortOrder) {
+		// oh ugh....
```

Comment:
Strong-typing mismatch: the upstream service call for keyset-paging returns a `TroubleCase`, but I need a `SneakyViewEntity` because we might need to sort on the `snoozeEnd` field, so I have to re-fetch from the DB.

Reply:
To close the loop: the final version will solve the problem rather than complaining about it. :-)